### DISCUSSION TITLE
`getModule`/`getReactModuleInfoProvider` required for TurboModules?

### DISCUSSION THREAD

**Discussion by lindboe** [reactwg/react-native-new-architecture#152] [Documentation]
I noticed in the "Migration to the New Architecture" > "Supporting in your Library" > "[Enabling in Android Library](https://reactnative.dev/docs/new-architecture-library-android)" section, there's no mention of the methods that are added to the ModulePackage class in [the guide to creating a new Turbo Module](https://reactnative.dev/docs/the-new-architecture/pillars-turbomodules#android).

The relevant example code:

```
- override fun getModule(name: String?, reactContext: ReactApplicationContext): NativeModule? = null
+ override fun getModule(name: String?, reactContext: ReactApplicationContext): NativeModule? =
+   if (name == CalculatorModule.NAME) {
+     CalculatorModule(reactContext)
+   } else {
+     null
+   }

- override fun getReactModuleInfoProvider() = ReactModuleInfoProvider? = null
+ override fun getReactModuleInfoProvider() = ReactModuleInfoProvider {
+   mapOf(
+     CalculatorModule.NAME to ReactModuleInfo(
+       CalculatorModule.NAME,
+       CalculatorModule.NAME,
+       false, // canOverrideExistingModule
+       false, // needsEagerInit
+       true, // hasConstants
+       false, // isCxxModule
+       true // isTurboModule
+     )
+   )
+ }
}
```

Is this step required to migrate to the New Architecture? If so, I think these docs should cover that and how to manage the existing Package class code (which I understand would be migrating away from using `createNativeModules`/`createViewManagers`, correct?)

**Comment by cortinico** [comment]
Hi @lindboe 
That's correct. The `*Package` class should also be updated to extend from `TurboReactPackage`.
Would you be up for sending a PR against the "[Enabling in Android Library](https://reactnative.dev/docs/new-architecture-library-android)" page to include a paragraph about it?

**Reply by lindboe** [reply]
Yup, I can do that, just wanted to make sure that's correct. Thanks!
